flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -762,7 +763,9 @@ private static PartitionInfo createPartitionInfo(
         ShuffleDescriptor shuffleDescriptor =
                 getConsumedPartitionShuffleDescriptor(
                         consumedPartition,
-                        TaskDeploymentDescriptorFactory.PartitionLocationConstraint.MUST_BE_KNOWN);
+                        TaskDeploymentDescriptorFactory.PartitionLocationConstraint.MUST_BE_KNOWN,
+                        // because partition is already finished, false is fair enough.
```

Comment:
Because `createPartitionInfo` will only be invoked from `finishPartitionsAndUpdateConsumers`, which has been already finished this partition. But in order not to be misused later, I renamed it to `createFinishedPartitionInfo`.